.github/gha_monitor/modules/github.py

```diff
@@ -0,0 +1,83 @@
+import json
+import logging
+from agithub import GitHub
+from datetime import datetime
+
+logger = logging.getLogger()
+
+
+class GitHub_Client:
+    # Over-ride
+    params = {
+        'github_username': None,
+        'secret': None,
+        'repo_organization': None,
+        'repo': None
+    }
+
+    def __init__(self):
+        assert self.params['repo_organization'] != None
+        assert self.params['repo'] != None
+
+        # Decide which auth method to use.
+        if 'github_username' in self.params and self.params['github_username'] != None:
+            assert self.params['github_password'] != None
+            self._github = GitHub.GitHub(self.params['github_username'], self.params['github_password'])
+            logging.debug('Using username/key to auth with Github: ' + self.params['github_username'])
+        elif 'token' in self.params and self.params['token'] != None:
+            logging.debug('Using a token to auth with GitHub')
+            self._github = GitHub.GitHub(token=self.params['token'])
+
+        self.response = {}
+        self.worklog = None
+        self.repo_org = self.params['repo_organization']
+        self.repo = self.params['repo']
+
+    def get_workflow_log_chunk(self, chunk=1, final_state='failure'):
+        """
+        Example using agithub:
+
+        username= get_user
+        secret = get_secret
+
+        client = GitHub.GitHub(username, secret)
+        client.repos.awslabs['private-s2n-fuzz'].actions.runs.get(page="1", status="failure")
+
+        :param final_state: str
+        :param chunk: int
+
+        """
+        (status_code, self.response) = \
+            self._github.repos[self.repo_org][self.repo].actions.runs.get(page=chunk,
+                                                                          status=final_state)
+        if status_code < 300:
+            self.worklog = GitHub_Worklog(self.response)
+            return True
+        else:
+            return False
+
+    def get_workflow_name(self, workflow_id):
+        logging.debug(f"Looking up workflow_id {workflow_id}")
+        (status_code, response) = \
+            self._github.repos[self.repo_org][self.repo].actions.workflows[workflow_id].get()
+        workflow_name = response['name']
+        logging.debug(f"Github workflow lookup gave us {workflow_name}")
+        if status_code < 300:
+            return workflow_name
+        else:
+            return None
```

Comment:
This type of return behavior is generally an antipattern.

http://docs.quantifiedcode.com/python-anti-patterns/maintainability/returning_more_than_one_variable_type_from_function_call.html

As with `get_workflow_chunk`, I would generally recommend either returning the value or raising an exception.

Reply:
implemented as well.